Filters don't (usually) determine the end of subscription (#1113)
moq-wg/moq-transport: draft-ietf-moq-transport.md

filter to avoid churn as downstream subscribers with disparate filters
subscribe and unsubscribe from a track.

A subscriber remains subscribed to a Track at a Relay until it unsubscribes or
the upstream publisher terminates the subscription. Subscription filters only

> allow a Relay to end the subscription when it knows that no additional Objects

Reviewer: I don't like text like this - it implies that relays needs to do this and that would be a challenging to implement in a way that did not have edge case problems across a multi relay networks.  I'd rather not have this unless we are going to exactly specify it.

Author: @fluffy I don't want to imply that relays are required to keep score and always end subscriptions this way, that's computationally infeasible.

Do you think we should prohibit it entirely?

> A subscriber remains subscribed to a Track at a Relay until it unsubscribes or
> the upstream publisher terminates the subscription. Subscription filters only
> allow a Relay to end the subscription when it knows that no additional Objects
> can match the filter.
> 

Reviewer 2: ```suggestion
A subscriber remains subscribed to a Track at a Relay until it unsubscribes or
its authorization expires or the upstream publisher terminates the subscription. 
Subscription filters only allow a Relay to end the subscription when it knows that 
no additional Objects can match the filter.
```

filter to avoid churn as downstream subscribers with disparate filters
subscribe and unsubscribe from a track.

> A subscriber remains subscribed to a Track at a Relay until it unsubscribes or
> the upstream publisher terminates the subscription, and all subgroups are
> accounted for. Subscriber filters do not affect when the subscription ends
> unless the Relay knows definitively that no additional Objects can arrive that
> match the filter.

Reviewer 2: I think i get the intent but I wonder if it helps reword it as below

```suggestion
Subscriber filters do not directly control the termination of a subscription. A subscription ends either when the subscriber unsubscribes or when the publisher terminates it..
```

Author: I want to leave the door open to a relay ending the subscription when everything is guaranteed delivered.  For example, subscribe range is all of group 10, and relay receives Group 10, Object 0, Status End of Group.

Reviewer 2: How about the updated text ?

> the upstream publisher terminates the subscription, and all subgroups are
> accounted for. Subscriber filters do not affect when the subscription ends
> unless the Relay knows definitively that no additional Objects can arrive that
> match the filter.

Reviewer 3: ```suggestion
the upstream publisher terminates the subscription. Subscription filters only
allow a publisher to end the subscription when the Relay knows that no 
additional Objects can match the filter.
```

Reviewer 3: A third suggestion :)

Author: I think you mean "only allow _the Relay_ to end the subscription when _it_" knows, otherwise looks great to me.

Reviewer 3: Updated.
